### PR TITLE
#294 Support dot notation for filter[...] parameter

### PR DESCRIPTION
**Pull request by remmeier**

### CODE REVIEW

File: katharsis-core/src/test/java/io/katharsis/queryspec/repository/DefaultQuerySpecDeserializerTest.java [hunk 1]
```diff
@@ -133,6 +133,18 @@ public void testFilterWithDefaultOp() throws InstantiationException, IllegalAcce
 	}
 
 	@Test
+	public void testFilterWithDotNotation() throws InstantiationException, IllegalAccessException {
+		QuerySpec expectedSpec = new QuerySpec(Task.class);
+		expectedSpec.addFilter(new FilterSpec(Arrays.asList("project", "name"), FilterOperator.EQ, "value"));
+
+		Map<String, Set<String>> params = new HashMap<>();
+		add(params, "filter[project.name]", "value");
```
Comment (chb0github) [inline katharsis-core/src/test/java/io/katharsis/queryspec/repository/DefaultQuerySpecDeserializerTest.java]: Please provide a test case that is several levels deep of `.` -- this forces edge cases out. 

It would be nice to validate the query against the actual object in question and throw a 400 if it can't be honored instead of just ignoring it

Comment (remmeier) [inline katharsis-core/src/test/java/io/katharsis/queryspec/repository/DefaultQuerySpecDeserializerTest.java]: done, 

the attribute names are actually already validated as long as that validation is not disabled.
